crud/update.lua

```diff
@@ -62,6 +62,12 @@ local function call_update_on_router(space_name, key, user_operations, opts)
         return nil, UpdateError:new("Wrong operations are specified: %s", err), true
     end
 
+    if type(key) == 'table' or type(key) == 'number' then
+        local tuple = space:get{key}
```

Comment:
You tries to get tuple on router but actually it is stored on storage. Also this construction is a bit wrong `{key}` -> `(key)`: otherwise it won't work for multipart indexes.

Also seems you've missed my comment that I shared you privately.
Update should be *optimistic*. Persons that perform correct update should get the same result - only one operation. If you get `box.error.NO_SUCH_FIELD_NO` or `box.error.NO_SUCH_FIELD_NAME` you should start to think about "adding intermediate fields" - try to go "get" tuple, add missing fields and then update.